### PR TITLE
chore: 11765 Migrate Merkle state classes to `swirlds-platform-core` module

### LINKED ISSUE #11765: Migrate `com.hedera.node.app.state` package to `swirlds-platform-core` module
Currently, `com.hedera.node.app.state` belongs to the `hedera-app` module. As a prerequisite for further refactorings, these classes have to be moved to two different modules: 
1. API classes (interfaces) should be moved to `platform-sdk/swirlds-state-api` module. Here is a list of classes to be moved:
```
HederaState

ReadableKVState
ReadableQueueState
ReadableSingletonState
ReadableStates
WritableKVState
WritableQueueState
WritableSingletonState
WritableStates

StoreMetrics
```

2.  Helper classes and base implementations of the API classes should be moved to `swirlds-platform-core` module (`com.swirlds.platform.state.merkle` package)

Here is a list of classes to be moved:
```
disk/
OnDiskKey
OnDiskKeySerializer
OnDiskReadableKVState
OnDiskValue
OnDiskValueSerializer
OnDiskWritableKVState

logging/
StateLogger

memory/
InMemoryKey
InMemoryReadableKVState
InMemoryValue
InMemoryWritableKVState

queue/
QueueNode
ReadableQueueStateImpl
WritableQueueStateImpl

singleton/
ReadableSingletonStateImpl
SingletonNode
StringLeaf
ValueLeaf
WritableSingletonStateImpl

./
StateUtils

```

This document provides more details about the change to be done - https://www.notion.so/swirldslabs/Public-Merkle-Tree-API-698ae50389b843cba04976ec0714fbd9

### PR DESCRIPTION
**Pull request by imalygin**
**Description**:

This PR moves classes related to Hedera state to `swirlds-platform-core` module. 

**Related issue(s)**:

Fixes #11765 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

File: hedera-node/test-clients/src/main/resource/log4j2.xml [hunk 1]
```diff
@@ -29,7 +29,7 @@
 	<Loggers>
 
 		<!-- Send transaction state logs to their own appender   -->
-		<Logger name="com.hedera.node.app.state.logging.TransactionStateLogger" level="info" additivity="false">
+		<Logger name="com.swirlds.platform.state.logging.TransactionStateLogger" level="info" additivity="false">
```
Comment (netopyr) [inline hedera-node/test-clients/src/main/resource/log4j2.xml:32]: Is this correct?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java [hunk 2]
```diff
@@ -309,15 +307,15 @@ public Hedera(@NonNull final ConstructableRegistry constructableRegistry) {
...
-                        FILE_SERVICE,
-                        new FreezeServiceImpl(),
-                        SCHEDULE_SERVICE,
-                        TOKEN_SERVICE,
-                        new UtilServiceImpl(),
-                        RECORD_SERVICE,
-                        BLOCK_SERVICE,
-                        FEE_SERVICE,
-                        CONGESTION_THROTTLE_SERVICE,
+                        // FILE_SERVICE,
```
[the remark this answers is not in the document]
Comment (imalygin) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:310]: Sorry, I was debugging, this got committed accidentally. I'll remove it

File: hedera-node/test-clients/src/main/resource/log4j2.xml (hunk 1 above)
Comment (imalygin) [reply hedera-node/test-clients/src/main/resource/log4j2.xml]: No. It's not. I removed it

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/MerkleTestBase.java [hunk 3]
```diff
@@ -16,40 +16,38 @@
 
 package com.hedera.node.app.state.merkle;
 
-import com.hedera.hapi.node.base.SemanticVersion;
-import com.hedera.node.app.spi.fixtures.state.StateTestBase;
+/*
```
[the remark this answers is not in the document]
Comment (imalygin) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/MerkleTestBase.java]: Removed it.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/MerkleHederaState.java [hunk 4]
```diff
@@ -766,4 +780,9 @@ public void remove(String stateKey) {
             queueInstances.remove(stateKey);
         }
     }
+
+    @NonNull
+    private static String extractStateKey(@NonNull StateMetadata md) {
```
[the remark this answers is not in the document]
Comment (imalygin) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/MerkleHederaState.java]: Done

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (imalygin) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java]: Done

**Review by edward-swirldslabs — APPROVED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/merkle/logging/StateLogger.java [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class StateLogger {
```
[the remark this answers is not in the document]
Comment (imalygin) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/merkle/logging/StateLogger.java:47]: Fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/merkle/singleton/SingletonNode.java [hunk 6]
```diff
@@ -46,10 +47,14 @@ public SingletonNode() {
...
 
-    public SingletonNode(@NonNull final StateMetadata<?, T> md, @NonNull final T value) {
-        setLeft(new StringLeaf(
-                StateUtils.computeLabel(md.serviceName(), md.stateDefinition().stateKey())));
-        setRight(new ValueLeaf<T>(md, value));
+    public SingletonNode(
+            @NonNull final String serviceName,
+            @NonNull String stateKey,
+            long classId,
+            Codec<T> codec,
```
[the remark this answers is not in the document]
Comment (imalygin) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/merkle/singleton/SingletonNode.java]: Fixed

**Review by edward-swirldslabs — APPROVED** [review]

**Review by tinker-michaelj — DISMISSED** [review]
Hmm not compiling.

**Review by Neeharika-Sompalli — COMMENTED** [review]
Need to resolve conflicts.

**Review by netopyr — APPROVED** [review]
LGTM - tyvm @imalygin

**Review by Neeharika-Sompalli — APPROVED** [review]
Changes LGTM. But need to resolve conflicts

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM!  Thanks @imalygin

**Review by tinker-michaelj — DISMISSED** [review]
LGTM, tyvm @imalygin !

**Review by netopyr — APPROVED** [review]
LGTM - tyvm @imalygin

**Review by timo0 — APPROVED** [review]
LGTM for base files

**Review by edward-swirldslabs — APPROVED** [review]

**Review by timo0 — APPROVED** [review]

**Review by david-bakin-sl — APPROVED** [review]
LGTM (reviewed only `cli-clients` and `hedera-smart-contract-service-impl`) - scary large PR turned out to be pretty vanilla in those two modules.

**Review by JeffreyDallas — APPROVED** [review]

**Review by netopyr — APPROVED** [review]
LGTM

**Review by Neeharika-Sompalli — APPROVED** [review]

**Review by david-bakin-sl — APPROVED** [review]
LGTM (same caveat: cli-clients and smart contracts only reviewed)

**Review by timo0 — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by JeffreyDallas — APPROVED** [review]

**Pull request merged**